geoadmin/src/main/java/org/activityinfo/geoadmin/model/ActivityInfoClient.java

```diff
@@ -252,4 +281,75 @@ public void postChildLevel(AdminLevel parentLevel, AdminLevel newLevel) {
     public ObjectMapper getObjectMapper() {
         return new ObjectMapperProvider().getContext(ObjectMapper.class);
     }
+    
+    
+    public FormClass getFormClass(int adminLevelId) {
+        
+        String json = formResource(CuidAdapter.adminLevelFormClass(adminLevelId)).path("class").get(String.class);
+
+
+        return FormClass.fromResource(Resources.fromJson(json));
+    }
+
+    public FormTree getFormTree(ResourceId resourceId) {
+        
+        if(localCatalog.isLocalResource(resourceId)) {
+            FormTreeBuilder treeBuilder = new FormTreeBuilder(localCatalog);
+            return treeBuilder.queryTree(resourceId);
+        
+        } else {
+
+            String json = formResource(resourceId).path("tree").get(String.class);
+            JsonObject object = new Gson().fromJson(json, JsonObject.class);
+            return JsonFormTreeBuilder.fromJson(object);
+        }
+    }
+
+    public FormTree getFormTree(int adminLevelId) {
+        return getFormTree(CuidAdapter.adminLevelFormClass(adminLevelId));
+    }
+    
+    public ColumnSet queryColumns(QueryModel queryModel) {
+        
+        if(localCatalog.isLocalQuery(queryModel)) {
+            ColumnSetBuilder builder = new ColumnSetBuilder(localCatalog, ColumnCache.NULL);
+            return builder.build(queryModel);
+
+        } else {
+
+            return queryColumnsRemotely(queryModel);
+        }
+    }
+
+    private ColumnSet queryColumnsRemotely(QueryModel queryModel) {
+        String json = client.resource(root)
+                .path("query")
+                .path("columns")
+                .type(MediaType.APPLICATION_JSON_TYPE)
+                .post(String.class, queryModel);
+
+        JsonObject object = new Gson().fromJson(json, JsonObject.class);
+        int numRows = object.getAsJsonPrimitive("rows").getAsInt();
+
+        Map<String, ColumnView> columnMap = new HashMap<>();
+        for (Map.Entry<String, JsonElement> column : object.getAsJsonObject("columns").entrySet()) {
+            JsonObject columnValue = column.getValue().getAsJsonObject();
+            String storage = columnValue.getAsJsonPrimitive("storage").getAsString();
+            switch (storage) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1301)